### PR TITLE
Add viewport dimension values to RUM session data

### PR DESCRIPTION
**Pull request by samuel-holt**
## Overview

Add the current browser's viewport width and height values to each session object that gets sent to the API. This uses `window.innerWidth` and `window.innerHeight` to get the dimensions, with a fallback to the `document.documentElement.clientWidth` and `document.documentElement.clientHeight` if the values are unavailable on the window object.

## Updates

- Add a separate class for handling access of the viewport from the globally available `window` and `document` objects (if available)
- Add the viewport dimension values to each session payload that gets sent in RUM
- Add a `trackViewportDimensions` option to disable the feature (defaults to true)
- Replace the existing `getViewport` function in CR with the new function

### CODE REVIEW

File: src/raygun.utilities/index.js [hunk 1]
```diff
@@ -211,18 +211,6 @@ window.raygunUtilityFactory = function(window, Raygun) {
       return Math.floor(Math.random() * 9007199254740993);
     },
 
-    getViewPort: function() {
-      if (this.isReactNative()) {
-        return { width: 'Not available', height: 'Not available' };
```
Comment (krishnaKapadia) [inline src/raygun.utilities/index.js:216]: Here the null object is defined like this with strings, but in the newer `getViewportDimensions` function we return nulls. Will this have negative impacts if the worker was relying on this "Not available" string for some reason?

**Review by krishnaKapadia — CHANGES_REQUESTED** [review]
Nice work Sam! I just have one important question :)

File: src/raygun.utilities/index.js (hunk 1 above)
Comment (samuel-holt) [reply src/raygun.utilities/index.js:216]: Good point, I decided to get rid of this as we want to move away from having workarounds for RN and migrate RN users to the new provider. I'll check if this value is important.

Comment (samuel-holt) [reply src/raygun.utilities/index.js:216]: I can't find any references to the "Not available" string, so the only change here would be that these values would now display as null in Crash Reporting

Comment (samuel-holt) [reply src/raygun.utilities/index.js:216]: Yep, that would be a good way to handle this - this is something we can do when we update the infrastructure :+1:

**Review by krishnaKapadia — DISMISSED** [review]
LG2M 🚀

**Review by JasenPalmer — APPROVED** [review]
Changes look good to me

**Pull request merged**